x-pack/plugins/remote_clusters/public/store/actions/remove_clusters.js

```diff
@@ -19,49 +19,75 @@ import {
 import { closeDetailPanel } from './detail_panel';
 import { getDetailPanelClusterName } from '../selectors';
 
+function getErrorTitle(count, name = null) {
+  if (count === 1) {
+    if (name) {
+      return i18n.translate('xpack.remoteClusters.removeAction.failureSingleNotificationTitle', {
+        defaultMessage: `Error removing remote cluster '{name}'`,
+        values: { name },
+      });
+    }
+  } else {
+    return i18n.translate('xpack.remoteClusters.removeAction.failureMultipleNotificationTitle', {
+      defaultMessage: `Error removing '{count}' remote clusters`,
+      values: { count },
+    });
+  }
+}
+
 export const removeClusters = (names) => async (dispatch, getState) => {
   dispatch({
     type: REMOVE_CLUSTERS_START,
   });
 
-  let removalSuccesses;
-  let removalErrors;
+  let itemsDeleted = [];
+  let errors = [];
 
   await Promise.all([
     sendRemoveClusterRequest(names.join(','))
       .then((response) => {
-        const { itemsDeleted, errors } = response.data;
-        removalSuccesses = itemsDeleted;
-        removalErrors = errors;
+        ({ itemsDeleted, errors } = response.data);
       }),
-    // Wait at least half a second to avoid a weird flicker of the saving feedback.
+    // Wait at least half a second to avoid a weird flicker of the saving feedback (only visible
+    // when requests resolve very quickly).
```

Comment:
It would be interesting to know why this app requires this hack and others don't. Let's leave it for now and you show it to me over zoom at some point. 😊

Reply:
Ideally we should do this anywhere we display a loading state and we care about a flicker disrupting the UX, though AFAIK it's inconsistently implemented. Here's how it looks without a minimum delay:

![async_feedback_flicker](https://user-images.githubusercontent.com/1238659/55743135-5ed6db00-59e6-11e9-9e28-387624be8592.gif)

Reply:
I see... it's kind of the React scheduler that we will get. I might have solved it differently to avoid adding a hack on each http request.
Wrap the full-screen loader with an HOC that sets a timeout of 300ms, and only after that time it shows the component. If it's less than 300 ms, don't do anything.

This way, all of Kibana benefit from it and we don't have to add Promise.all.